app/models/spree/order_decorator.rb

```diff
@@ -31,20 +32,56 @@ def create_mailchimp_cart
         update_column(:mailchimp_cart_created, true)
       end
 
+      def after_ship_jobs
+        order_shipped_notification
+        update_mailchimp_order
+      end
+
+      def after_refund_jobs
+        order_refunded_notification
+        update_mailchimp_order
+      end
+
       private
 
       def after_create_jobs
+        new_order_notification
         create_mailchimp_order
         delete_mailchimp_cart
       end
 
+      def after_cancel_jobs
+        order_canceled_notification
+        update_mailchimp_order
+      end
+
       def delete_mailchimp_cart
         ::SpreeMailchimpEcommerce::DeleteCartJob.perform_later(number)
       end
 
       def create_mailchimp_order
         ::SpreeMailchimpEcommerce::CreateOrderJob.perform_later(mailchimp_order)
       end
+
+      def update_mailchimp_order
+        ::SpreeMailchimpEcommerce::UpdateOrderJob.perform_later(self)
+      end
+
+      def new_order_notification
+        @notification = ::SpreeMailchimpEcommerce::Presenters::OrderNotificationPresenter.new(self).invoice_or_order_confirmation
```

Comment:
also, let's avoid to use instance variables here. I suggest to create a `notification` method returns proper notification. we want to have only one place where we define it.